src/swapRouter.ts

```diff
@@ -230,7 +411,9 @@ export abstract class SwapRouter {
     }
 
     const numberOfTrades = trades.reduce(
-      (numberOfTrades, trade) => numberOfTrades + (trade instanceof V3Trade ? trade.swaps.length : 1),
+      // TODO: is this math right for mixedRouteTrade?
+      (numberOfTrades, trade) =>
+        numberOfTrades + (trade instanceof V3Trade || trade instanceof MixedRouteTrade ? trade.swaps.length : 1),
```

Comment:
Not really sure what the purpose of this math is, and what the rules should be for MixedRouteTrades